File: 8-leetcode/021-merge-two-lists.rb
# Definition for singly-linked list.
# class ListNode
#     attr_accessor :val, :next
#     def initialize(val)
#         @val = val
#         @next = nil
#     end
# end

# @param {ListNode} l1
# @param {ListNode} l2
# @return {ListNode}

def merge_two_lists(l1, l2)    
  if l1 && l2
    if l1.val >= l2.val
      l1, l2 = l2, l1
    end
      l1.next = merge_two_lists(l1.next, l2)
  end
  l1.nil? ? l2 : l1
end

File: 8-leetcode/021-merge-two-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var mergeTwoLists = (left, right) => {
  if (left && right) {
    if (left.val >= right.val) {
      let temp = left;
      left = right;
      right = temp;
    }
    left.next = mergeTwoLists(left.next, right);
  }
  return (left === null ? right : left)
}